### ISSUE TITLE
Add gzip support

### ISSUE THREAD

**Issue by Altai-man** [croservices/cro-http#130]
This is now de facto required to serve a website nicely, so why not support this? Yes, an nginx proxy can do this instead, but I don't think we want to rely on solutions of others forever and ever.

**Comment by jnthn** [comment]
Agree. It's at least as important in the HTTP client too, where "stick Apache/nginx in front" is not an option.

**Comment by Zer0-Tolerance** [comment]
Hi there, I would vote for this feature as well !
